tensorflow/core/tpu/tpu_initializer_helper.cc

```diff
@@ -51,6 +54,62 @@ bool GetEnvBool(const char* name, bool defval) {
 
 }  // namespace
 
+bool PrintFdMessage(long tgid) {
+  char path[40], line[100];
+  DIR* fd;
+  struct dirent* ent;
+  long deviceId;
+  snprintf(path, 40, "/proc/%ld/fd", tgid); 
+  fd = opendir(path);
+  if(!fd) {
+    return false;
+  }
+  while(ent = readdir(fd)) {
+    if(!isdigit(*ent->d_name))
+      continue;
+    deviceId = strtol(ent->d_name, NULL, 10);
+    char paths[40];
+    int statusfd;
+
+    snprintf(paths, 40, "/proc/%ld/fd/%ld", tgid, deviceId);
+    statusfd = readlink(paths, line, 100);
+    if (!statusfd)
+      return false;
+    if(strncmp(line, "/dev/accel0",11) != 0)
```

Comment:
Along with switching to using absl, you should use std::strings, which I think you can just use == with (google this :P)